cli/cmd/plugin/ui/web/tanzu-ui/src/views/workload-cluster/WorkloadClusterUtility.tsx

```diff
@@ -0,0 +1,52 @@
+import { ManagementCluster } from '../../shared/models/ManagementCluster';
+import { ChangeEvent } from 'react';
+
+// Utility method for retrieving a value from a change event.
+// Returns true/false from checkboxes; otherwise the value in the event object
+export function getValueFromChangeEvent(evt: ChangeEvent<HTMLSelectElement>) {
+    let value
+    if (evt.target.type === 'checkbox') {
+        // @ts-ignore
+        value = evt.target['checked']
+    } else {
+        value = evt.target.value
+    }
+    return value
+}
+
+// NOTE: Rather than store the CC variables at the top level of our data store,
+//       we accumulate them in a single object per management cluster.
+//       This makes using them much easier (later), and prevents "collisions" if the user switches management clusters.
+function getClusterVariableData(managementCluster: ManagementCluster, state: any): any {
+    if (!managementCluster || !managementCluster.name) {
+        console.error('getClusterVariableData() called with undefined/unnamed cluster')
+        return {}
+    }
+    const ccVarData = { ...state.data.CLUSTER_CLASS_VARIABLE_VALUES }
+    const ccVarClusterData = ccVarData[managementCluster.name]
+    return ccVarClusterData ? ccVarClusterData : {}
```

Comment:
If you wanted to get rid of the expression, I thiiiiink you can just `return ccVarData[managementCluster.name] || {}` which should return the empty object if ccVarData[...] is undefined